db/migrate/20250519010827_migrate_data_to_push_model.rb

```diff
@@ -0,0 +1,223 @@
+class MigrateDataToPushModel < ActiveRecord::Migration[7.2]
+  def up
+    # Migrate data from Password model to Push model
+    migrate_passwords
+    
+    # Migrate data from FilePush model to Push model
+    migrate_file_pushes
+    
+    # Migrate data from Url model to Push model
+    migrate_urls
+  end
+  
+  def down
+    attach_files_to_old_records
+  end
+  
+  private
+
+  def attach_files_to_old_records
+    Push.where(kind: "file", expired: false).find_each do |push|
+      begin 
+        file_push = FilePush.find_by(url_token: push.url_token)
+        if file_push
+          push.files.attachments.each do |attachment|
+            attachment.update!(record_type: "FilePush", record_id: file_push.id)
+          end
+        else
+          puts "FilePush not found for push #{push.id} and url_token #{push.url_token}"
+        end
+      rescue => e
+        puts "Error migrating attachments of push #{file_push.id} having 'file' kind: #{e.message}"
+      end
+    end
+  end
+  
+  # Password migration methods
+  def migrate_passwords
+    puts "Migrating passwords to pushes..."
+    
+    Password.where(expired: false).find_each do |password|
```

Comment:
I just tried the migration on test database and it works great (including file attachments).  The one issue is that we should also migrate `expired` Passwords, FilePushes and Urls too so that users still have their historical data.

Reply:
It is updated to migrate `expired` old records. Necessary tests are added.